internal/provider/resource_vault_cluster.go

```diff
@@ -823,7 +869,21 @@ func setVaultClusterResourceData(d *schema.ResourceData, cluster *vaultmodels.Ha
 	if cluster.PerformanceReplicationInfo != nil {
 		prInfo := cluster.PerformanceReplicationInfo
 		if prInfo.PrimaryClusterLink != nil {
-			primaryLink, err := linkURL(cluster.PerformanceReplicationInfo.PrimaryClusterLink)
+			primaryClusterLink := &sharedmodels.HashicorpCloudLocationLink{
```

Comment:
can we consider moving this translation to another func? that will allow us to re-use this in other places.